fix OperationId for autorest.powershell m4

TheOnlyWei opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify: breaking change in autorest.powershell m4 requires changes in our specs.
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. TBA
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. TBA
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

TheOnlyWei committed:
update azsadmin specification

TheOnlyWei commented:
The reason for this change is because autorest.powershell m3 is no longer being maintained and we have to update to autorest.powershell m4, which has a breaking change that requires updating all of our OperationId properties from something like:

```
        "tags": [
          "Subscriptions"
        ],
        "operationId": "UpdateEncryption",
```

To the following:

```
        "tags": [
          "Subscriptions"
        ],
        "operationId": "Subscriptions_UpdateEncryption",
```

bganapa commented:
@ArcturusZhang COuld you pls help to see why this LintDiff is failing. it seems to be an issue with the tool itself..
Please note that this is for AzureStack Admin APIs . These APIs are NOT relevant for public azure 
https://github.com/Azure/azure-rest-api-specs/pull/13283/checks?check_run_id=2036845597
"readme":"azsadmin/resource-manager/subscriptions/readme.md",
"tag":"package-2015-11-01",
"details":"Error: Could not read 'file:///home/vsts/work/1/azure-rest-api-specs/specification/profiles/readme.md'."

ArcturusZhang commented:
@TheOnlyWei @bganapa changes of the operationIds cause a lot of breaking changes in the downstream SDKs.
The readme files in the impacted swaggers said we have only C# SDK configured for those swaggers. Therefore at least we need to confirm the breaking change impacts on dot net SDK. Could you please confirm?

bganapa commented:
> @TheOnlyWei @bganapa changes of the operationIds cause a lot of breaking changes in the downstream SDKs.
> The readme files in the impacted swaggers said we have only C# SDK configured for those swaggers. Therefore at least we need to confirm the breaking change impacts on dot net SDK. Could you please confirm?
@ArcturusZhang We do not ship any C# SDK or any other language SDK for these affected AzureStack Operator APIs, we ship only Powershell, which is not going to have any breaking change.


The change is mandated by autorest modeller changes, as you can see only for few operations the spec is not following group_operation format. 
We have the powershell generated with autorest.powershell 

Linking an issue which discusses why this is needed
https://github.com/Azure/autorest.powershell/issues/741

jianyexi commented:
@bganapa Seems there is missing tag statement at line:110  file 
https://github.com/Azure/azure-rest-api-specs/blob/7b3f9ce2b4934faa27c480de119676f9da09849f/specification/azsadmin/resource-manager/subscriptions/readme.md#tag-package-2015-11-01

The right writing is the below : 
` ``` yaml $(tag) == 'all-api-versions'`

bganapa commented on specification/azsadmin/resource-manager/subscriptions/readme.md:
````diff
@@ -107,7 +107,7 @@ AutoRest V3 generators require the use of `--tag=all-api-versions` to select api
 This block is updated by an automatic script. Edits may be lost!
 
-``` yaml
+``` yaml $(tag) == 'all-api-versions
````
single quote missing

TheOnlyWei replied:
I added that quote, but forgot to push the change. I have updated it now.

TheOnlyWei commented:
Hi @jianyexi, I have added a commit that added the tag.

ArcturusZhang reviewed (approved):
I have no objections since this change will not affect any downstream SDKs.

ArcturusZhang commented:
@TheOnlyWei please let me know when this is ready to merge.

ArcturusZhang commented:
@TheOnlyWei @jianyexi could you please take a look at the LintDiff error? Do we need to ignore these or fix them?

bganapa commented:
Model definition 'SubscriptionDefinition' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
Location: Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json#L770

This is expected as Subscription is not defined as a ARM resource (even for azure). This should be exempted

TheOnlyWei commented:
Hi @ArcturusZhang @jianyexi , do you have any ideas what the new Swagger pipeline errors are about? There are some internal_errors and links to "undefined" section of our code that lead to 404 pages.

ArcturusZhang commented on specification/azsadmin/resource-manager/subscriptions/readme.md:
````diff
@@ -107,7 +107,7 @@ AutoRest V3 generators require the use of `--tag=all-api-versions` to select api
 
 This block is updated by an automatic script. Edits may be lost!
 
-``` yaml
+``` yaml $(tag) == 'all-api-versions'
 # include the azure profile definitions from the standard location
 require: $(this-folder)/../../../../profiles/readme.md
 
````
We could remove this whole section of "Multi-API/Profile support for AutoRest v3 generators"
It is auto-generated previously and now deprecated, we do not need to maintain this section.

ArcturusZhang reviewed (commented):
I am seeing that the SDK automation pipeline is also not work as expected (previously we have a result for the powershell geneation and now it is gone). @PhoenixHe-msft to help on this.

PhoenixHe-msft commented:
swagger-to-sdk section is missing in subscriptions/readme.md, and commerce/readme.md has wrong config in swagger-to-sdk with azure-sdk-for-sdk. Please fix the readme.md.

bganapa commented:
@PhoenixHe-msft We do not ship any language SDK for these APIs (AzureStack Administrator APIs). These APIs do not exist in azure.

Do we still need swagger-to-sdk section? if so could you pls point us to a sample? Thank you

PhoenixHe-msft commented:
@bganapa In that case please remove swagger-to-sdk and add comment in readme.md. OK to do that.

TheOnlyWei commented:
Hi @ArcturusZhang I removed the "Multi-API/Profile support for AutoRest v3 generators" section.

Hi @PhoenixHe-msft, I have removed the swagger-to-sdk from commerce readme and added comment about not needing swagger-to-sdk:

But the same errors are still there.

ArcturusZhang commented:
Add @raych1 @jianyexi for help on the failures of ModelValidation, LintDiff and SemanticValidation

bganapa commented:
@TheOnlyWei  Could you pls start a mail thread so that we could find out the issue here and merge this sooner?

@raych1 @jianyexi pls let us know if you have looked in to the errors in this PR. Thanks!

raych1 commented on specification/azsadmin/resource-manager/subscriptions/Microsoft.Subscriptions.Admin/preview/2015-11-01/Subscriptions.json:
```diff
@@ -444,6 +547,7 @@
             }
           }
         },
+        "isDataAction": false,
```
`isDataAction` cannot set `false` here. It should annotated as an object with type.

TheOnlyWei replied:
Hi @raych1, do you have an example of the object? I remember reading the readme rules help guide regarding this change and "false" is the example they gave as the correct value. I can't find it now since it was an error from last pipeline build.

TheOnlyWei commented:
Hi @bganapa, can you take a quick look over the changes and see that I did not change anything that should be an exception and not be changed?

Hi @ArcturusZhang, can I get an exception for the following errors? Our current Azure Stack modules expect the REST api properties that we have currently and we cannot change that, but they are showing up as autorest errors in the Swagger LintDiff build.

R2020 - RequiredPropertiesMissingInResourceModel
R3006 - BodyTopLevelProperties
R4017 - TopLevelResourcesListBySubscription
R4017 - TopLevelResourcesListBySubscription

There is also an autorest exception that I am not sure what is the cause of. @raych1 do you have any ideas? I did not get this error running the command in the error locally.

```
2021-04-29T22:56:25.8169066Z Processing "specification/azsadmin/resource-manager/subscriptions/readme.md":
2021-04-29T22:56:25.8207964Z Executing: npx autorest --validation --azure-validator --semantic-validator=false --model-validator=false --message-format=json --openapi-type=arm --use=@microsoft.azure/classic-openapi-validator@1.1.6 --use=@microsoft.azure/openapi-validator@1.9.2 --tag=package-2015-11-01 specification/azsadmin/resource-manager/subscriptions/readme.md
2021-04-29T22:56:29.4277212Z Failed to acquire 'require'd configuration 'file:///home/vsts/work/1/c93b354fd9c14905bb574a8834c4d69b/specification/profiles/readme.md'
2021-04-29T22:56:29.4278709Z   Error: Could not read 'file:///home/vsts/work/1/c93b354fd9c14905bb574a8834c4d69b/specification/profiles/readme.md'.
```

bganapa commented on specification/azsadmin/resource-manager/commerce/Microsoft.Commerce.Admin/preview/2015-06-01-preview/examples/Operations/List.json:
```diff
@@ -17,6 +22,11 @@
           },
           {
             "name": "Microsoft.Commerce.Admin/operations/read",
+            "isDataAction": {
```
In the examples it should simply be isDataAction: false

bganapa reviewed (changes requested):
minor change in the examples

bganapa commented on specification/azsadmin/resource-manager/azurebridge/Microsoft.AzureBridge.Admin/preview/2016-01-01/AzureBridge.json:
```diff
@@ -92,11 +92,7 @@
           "$ref": "#/definitions/Display",
           "readOnly": true
         },
-        "isDataAction": {
```
@TheOnlyWei  The chenge needs to be reverted only under the examples folder. we need to keep it here

bganapa reviewed (changes requested):
minor changes requested
